divi/src/kernel.cpp

```diff
@@ -310,7 +310,7 @@ bool CheckCoinstakeForVaults(const CTransaction& tx, const CBlockRewards& expect
                              const CCoinsViewCache& view)
 {
     if (!tx.IsCoinStake())
-        return error("%s: transaction is not a coinstake", __func__);
+        return true;
```

Comment:
Revert these. If it's not a coinstake it should fail. I realize other parts will fail the block anyways but in refactoring main we don't want the "decentralization of responsability" for validation as that will make it substantially more difficult later

Reply:
See other comments.  I think this change makes sense, and should have been done earlier already like that.  But of course if you prefer to have this in a separate PR (or commit), I can do so, but I feel that is not necessary in this case.  And if you disagree about the change in general and believe the previous version is what it should be (independent of the particular PR here), let me know and I can of course change it back.